go-controller/pkg/ovn/egressgw_test.go

```diff
@@ -1952,6 +1954,69 @@ var _ = ginkgo.Describe("OVN Egress Gateway Operations", func() {
 			err := app.Run([]string{app.Name})
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
+		ginkgo.It("should create a single policy for concurrent addHybridRoutePolicy for the same node", func() {
+			app.Action = func(ctx *cli.Context) error {
+				config.Gateway.Mode = config.GatewayModeLocal
+
+				fakeOvn.startWithDBSetup(
+					libovsdbtest.TestSetup{
+						NBData: []libovsdbtest.TestData{
+							&nbdb.LogicalRouterPort{
+								UUID:     ovntypes.GWRouterToJoinSwitchPrefix + ovntypes.GWRouterPrefix + "node1" + "-UUID",
+								Name:     ovntypes.GWRouterToJoinSwitchPrefix + ovntypes.GWRouterPrefix + "node1",
+								Networks: []string{"100.64.0.4/32"},
+							},
+							&nbdb.LogicalRouter{
+								Name: ovntypes.OVNClusterRouter,
+								UUID: ovntypes.OVNClusterRouter + "-UUID",
+							},
+						},
+					},
+				)
+				intPriority, _ := strconv.Atoi(types.HybridOverlayReroutePriority)
+				finalNB := []libovsdbtest.TestData{
+					&nbdb.LogicalRouterPolicy{
+						UUID:     "2a7a61cb-fb13-4266-a3f0-9ac5c4471123 [u2596996164]",
+						Priority: intPriority,
+						Action:   nbdb.LogicalRouterPolicyActionReroute,
+						Nexthops: []string{"100.64.0.4"},
+						Match:    "inport == \"rtos-node1\" && ip4.src == $a17568862106095406051 && ip4.dst != 10.128.0.0/14",
+					},
+					&nbdb.LogicalRouter{
+						Name:     ovntypes.OVNClusterRouter,
+						UUID:     ovntypes.OVNClusterRouter + "-UUID",
+						Policies: []string{"2a7a61cb-fb13-4266-a3f0-9ac5c4471123 [u2596996164]"},
+					},
+					&nbdb.LogicalRouterPort{
+						UUID:     ovntypes.GWRouterToJoinSwitchPrefix + ovntypes.GWRouterPrefix + "node1" + "-UUID",
+						Name:     ovntypes.GWRouterToJoinSwitchPrefix + ovntypes.GWRouterPrefix + "node1",
+						Networks: []string{"100.64.0.4/32"},
+					},
+				}
+
+				g := new(errgroup.Group)
+				c := make(chan int)
+				for i := 1; i <= 5; i++ {
+					podIndex := i
+					g.Go(func() error {
+						<-c
+						return fakeOvn.controller.addHybridRoutePolicyForPod(net.ParseIP(fmt.Sprintf("10.128.1.%d", podIndex)), "node1")
+					})
+				}
+				close(c)
+				err := g.Wait()
+				gomega.Expect(err).To(gomega.HaveOccurred()) // we expect a timeout err from ovsdb
```

Comment:
Is this chance based on the 5 attempts that we are making above? Or are we guaranteed to get an error? Always a timeout error or can we get some other type of error? Trying to rule out that this will fail at some later point in time, it smells a bit like it may happen.

Reply:
after talking on slack, not checking this err anymore as we are only interested that there is a single policy in the database and not on whatever is returned here.
thanks